var/spack/repos/builtin/packages/hermes/package.py

```diff
@@ -11,38 +11,136 @@ class Hermes(CMakePackage):
+        url="https://github.com/HDFGroup/hermes/archive/refs/tags/v1.0.5-beta.tar.gz",
+        sha256="1f3ba51a8beda4bc1314d6541b800de1525f5e233a6f498fcde6dc43562ddcb7",
+    )
+    version(
+        "1.0.0-beta",
+        url="https://github.com/HDFGroup/hermes/archive/refs/tags/v1.0.0-beta.tar.gz",
+        sha256="301084cced32aa00532ab4bebd638c31b0512c881ffab20bf5da4b7739defac2",
+    )
+    version(
+        "0.9.9-beta",
+        url="https://github.com/HDFGroup/hermes/archive/refs/tags/v0.9.9-beta.tar.gz",
+        sha256="d2e0025a9bd7a3f05d3ab608c727ed15d86ed30cf582549fe996875daf6cb649",
+    )
+    version(
+        "0.9.8-beta",
+        url="https://github.com/HDFGroup/hermes/archive/refs/tags/v0.9.8-beta.tar.gz",
+        sha256="68e9a977c25c53dcab7d7f6ef0df96b2ba4a09a06aa7c4a490c67faa2a78f077",
+    )
+    version(
+        "0.9.5-beta",
+        url="https://github.com/HDFGroup/hermes/archive/refs/tags/v0.9.5-beta.tar.gz",
+        sha256="f48d15591a6596e8e54897362ec2591bc71e5de92933651f4768145e256336ca",
+    )
     version(
         "0.9.0-beta",
         url="https://github.com/HDFGroup/hermes/archive/refs/tags/v0.9.0-beta.tar.gz",
         sha256="abf258a52fa79729dfeb28559957abf8945f3ad37cadefb3bc685227c5f057a8",
     )
 
-    depends_on("c", type="build")  # generated
-    depends_on("cxx", type="build")  # generated
-
+    variant("adios", default=False, description="Build Adios tests")
+    variant("ares", default=False, description="Enable full libfabric install")
+    variant("compress", default=False, description="Enable compression")
+    variant("encrypt", default=False, description="Enable encryption")
+    variant("mpiio", default=True, description="Enable MPI I/O adapter")
+    variant("python", default=False, description="Build Python Wrapper")
+    variant("stdio", default=True, description="Enable STDIO adapter")
     variant("vfd", default=False, description="Enable HDF5 VFD")
+    variant("zmq", default=False, description="Build ZeroMQ tests")
+
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
+    depends_on("pkgconfig", type="build")
+    depends_on("libelf")
+
+    depends_on("hermes-shm@master+boost+cereal+mochi")
+
+    depends_on("hermes-shm+adios", when="+adios")
+    depends_on("hermes-shm+ares", when="+ares")
+    depends_on("hermes-shm+compress", when="+compress")
+    depends_on("hermes-shm+encrypt", when="+encrypt")
+    depends_on("hermes-shm+mpiio", when="+mpiio")
+    depends_on("hermes-shm+vfd", when="+vfd")
+    depends_on("hermes-shm+zmq", when="+zmq")
 
-    depends_on("mochi-thallium~cereal@0.8:")
-    depends_on("catch2")
-    depends_on("glpk")
-    depends_on("glog@0.4.0:")
-    depends_on("mpi")
-    depends_on("hdf5@1.14.0:", when="+vfd")
-    depends_on("yaml-cpp")
+    depends_on("py-jarvis-util", type="test")
+
+    depends_on("mpi", when="+mpiio")
+    conflicts("^[virtuals=mpi] nvhpc", when="+mpiio", msg="+mpio does not support nvhpc MPI")
```

Comment:
`depends_on("mpi")` -> `depends_on("mpi", when="+mpiio")` - Is needed to get the selected virtual for `"mpi"` in spack.

The conflict with `nvhpc` is needed as the selection which, variant to use does not have support for `"mpi" from a build of `nvhpc+mpi`, and if one such build already exists, and none for `openmpi` or `mpich`, the `concretizer` does select `nvhpc` for `"mpi"` if  `--reuse` (default) is in effect.